packages/utilities/src/lib/poll.ts

```diff
@@ -0,0 +1,49 @@
+import { sleep } from './sleep';
+import type { Awaitable } from './types';
+
+/**
+ * Executes a function {@link cb} and validates the result with function {@link cbCondition},
+ * and repeats this until {@link cbCondition} returns `true` or the {@link timeoutMilliseconds} is reached.
+ *
+ * For a synchronous variant, see [pollSync](./pollSync.d.ts).
+ * @param cb The function that should be executed.
+ * @param cbCondition A function that when given the result of `fn` should return `true` if the polling should stop and should return `false` if the polling should continue.
+ * @param intervalMilliseconds The interval in between each poll.
+ * @param timeoutMilliseconds The maximum amount of time to poll before timing out and throwing an error.
+ * @param verbose Whether to log to the console on each polling interval, allowing the tracing of the amount of required attempts.
+ * @returns The result of {@link cb} as soon as {@link cbCondition} returns `true`, or an error if {@link timeoutMilliseconds} is reached.
+ * @throws If {@link timeoutMilliseconds} is reached.
+ */
+export async function poll<T>(
+	cb: () => Awaitable<T>,
+	cbCondition: (arg: Awaitable<T>) => boolean,
```

Comment:
You want the unwrapped `Promise<T>`:
```suggestion
	cbCondition: (arg: Awaited<T>) => boolean,
```